### PR TITLE
Implemented print and println delegates to pass through the command's context

### LINKED ISSUE #27: `print` and `println` instead of `echo` on CliktCommand
After having built some CLI's lately using the library, I'm wondering if it would be beneficial to just expose `print` and `println` methods to mimic printing to the screen directly with `System.out` or `System.err` but in scope with the console's Context instead

Example proposed methods:
```kotlin
protected fun println() = echo("")
protected fun println(message: String) = echo(message)
protected fun println(message: Any?) = echo(message?.toString())

protected fun print(message: String) = echo(message, trailingNewLine = false)
protected fun print(message: Any?) = echo(message?.toString(), trailingNewLine = false)
```

We could also consider just exposing a `out` and `err` PrintStream on the CliktCommand type 

```kotlin
class CliktCommand(...) {
    protected val out: PrintStream by lazyPrintStream { echo(it, trailingNewLine = false) }
    protected val err: PrintStream by lazyPrintStream { echo(it, trailingNewLine = false, err = true) }

    ...

    private inline fun lazyPrintStream(output: (String) -> Unit): Lazy<PrintStream> = lazy(LazyThreadSafetyMode.NONE) {
        PrintStream(object : OutputStream() {
            override fun write(b: Int) {
                output(b.toByte().toChar().toString())
            }
        })
    }
}
```

at which point we could just alias the recommended methods above to go to `out` by default

```kotlin
protected fun println() = out.println()
protected fun println(message: String) = out.println(message)
protected fun println(message: Any?) = out.println(message?.toString())

protected fun print(message: String) = out.print(message)
protected fun print(message: Any?) = out.print(message?.toString())
```

### PR DESCRIPTION
**Pull request by bradynpoulsen**
Closes #27

### CODE REVIEW

File: clikt/src/main/kotlin/com/github/ajalt/clikt/output/CliktConsole.kt [hunk 1]
```diff
@@ -38,11 +38,8 @@ interface CliktConsole {
     val lineSeparator: String
 }
 
-class InteractiveCliktConsole(private val console: Console) : CliktConsole {
-    override fun promptForLine(prompt: String, hideInput: Boolean) = when {
-        hideInput -> console.readPassword(prompt)?.let { String(it) }
-        else -> console.readLine(prompt)
-    }
+abstract class SystemCliktConsole internal constructor(): CliktConsole {
```
Comment (ajalt) [inline clikt/src/main/kotlin/com/github/ajalt/clikt/output/CliktConsole.kt]: :+1:

**Comment by ajalt** [comment]
After some discussion, I don't think it's a good idea to override the built-in functions by default. However, we could still provide this functionality as an opt-in feature. By doing something like this:

Make an interface that defines the `ConsoleStream`s

```kotlin
interface ConsoleStreamProvider {
    val out : ConsoleStream
    val err : ConsoleStream
}
```

And have `CliktCommand` implement that. Then you can define another interface with the print methods and default implementations:

```kotlin
interface TermUiPrint: ConsoleStreamProvider {
    fun println(message: Any?) = out.println(message)
    fun print(message: Any? = "\n") = out.print(message)
}
```

Then you could opt-in to the functionality by using `TermUiPrint` as a mixin on you commands.

**Comment by jcornaz** [comment]
> After some discussion, I don't think it's a good idea to override the built-in functions by default. However, we could still provide this functionality as an opt-in feature. 

@ajalt Isn't `echo` already an opt-in?

If I want to use kotlin's behaviour I use `print` or `println`. If I want to have clikt's behaviour I use `echo`.

**Comment by ajalt** [comment]
Yes, I agree. Apologies for changing my mind on this, but I'm going to close the PR. Thank you for your contributions!

**Comment by bradynpoulsen** [comment]
Yeah, ultimately, it doesn't really seems to be worth including in the base library unless it's part of the core functionality to provide `println` similar to how `require` is provided

**Pull request closed without merging**